### PR TITLE
Python: Improve the PAM authentication bypass query

### PR DESCRIPTION
**Pull request by porcupineyhairs**
The current PAM auth bypass query which was contributed by me a few months back, alert on a vulenrable function but does not check if the function is actually function. This leads to a lot of fasle positives.

With this PR, I add a taint-tracking configuration to check if the username parameter can actually be supplied by an attacker.

This should bring the FP's significantly down.

CC: @RasmusWL

### CODE REVIEW

**Comment by RasmusWL** [comment]
Thanks for your submission @porcupineyhairs, sorry for the late reply, my life has been kinda hectic :zap: :sleeping:  

I agree with the general idea behind these query changes, it will be more accurate if we can track user-controlled data :+1:

However, since this query is now promoted to be part of our standard query suite, the code quality will also need to match that of our other queries. So you would need to follow the structure of our other path-problems, which looks like this (using SQL injection as an example):

- https://github.com/github/codeql/blob/b3f10311b3e70dcbceeb045479263b83a5c1f4d9/python/ql/src/Security/CWE-089/SqlInjection.ql
- https://github.com/github/codeql/blob/b3f10311b3e70dcbceeb045479263b83a5c1f4d9/python/ql/lib/semmle/python/security/dataflow/SqlInjectionQuery.qll
- https://github.com/github/codeql/blob/b3f10311b3e70dcbceeb045479263b83a5c1f4d9/python/ql/lib/semmle/python/security/dataflow/SqlInjectionCustomizations.qll

(as always, ensure proper QLDocs and auto formatting)

**Review by RasmusWL — APPROVED** [review]
Thanks @porcupineyhairs :+1: I took the liberty of fixing up a few other small things, so now it looks good to me :+1:

I might get some more :eyes: on the select text, but other than that, I don't see anything holding us back from merging this.

_Please don't make more force pushes to the PR, so we can retain the history :blush:_

**Comment by RasmusWL** [comment]
> @RasmusWL Sorry about that. The force push was due to a bunch of unrelated work getting commit with this one. By mistake, I had added about 80-90 go files.

no problem. I just meant from now on. My bad if that was unclear.

File: python/ql/lib/change-notes/2022-11-17-py-pam-improve.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
 ---
  category: majorAnalysis
 ---
-* Converted `py/pam-auth-bypass` to a data-flow query, resulting in significantly lower false positives.
+* The _PAM authorization bypass due to incorrect usage_ (`py/pam-auth-bypass`) query has been converted to a tain-tracking query, resulting in significantly fewer false positives.
```
[the remark this answers is not in the document]
Comment (RasmusWL) [reply python/ql/lib/change-notes/2022-11-17-py-pam-improve.md:4]: yes, it's for making things italic, such as _this_ example.

File: python/ql/lib/change-notes/2022-11-17-py-pam-improve.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+ category: majorAnalysis
+---
+* The _PAM authorization bypass due to incorrect usage_ (`py/pam-auth-bypass`) query has been converted to a tain-tracking query, resulting in significantly fewer false positives.
```
Comment (intrigus-lgtm) [inline python/ql/lib/change-notes/2022-11-17-py-pam-improve.md:4]:
```suggestion
* The _PAM authorization bypass due to incorrect usage_ (`py/pam-auth-bypass`) query has been converted to a taint-tracking query, resulting in significantly fewer false positives.
```

**Review by RasmusWL — APPROVED** [review]

**Pull request merged**